### PR TITLE
Front notification

### PR DESCRIPTION
**Pull request by Mamar-dev**
Add web socket notification

### CODE REVIEW

File: src/front/src/pages/home/Home.tsx [hunk 1]
```diff
@@ -2,12 +2,22 @@ import { Outlet } from "react-router-dom";
...
 const BACKEND_PORT = 8080
 
 declare interface WebSocketListener {
     on(event: 'charge-point-update', listener: (message: WebSocketChargePointNotification) => void): this;
+    on(event: 'notify', listener: (message: NotificationMessage) => void): this;
+}
+
+type WebSocketMessage = {
+  name: "ChargePointWebsocketNotification" | "CriticalityWebsocketNotification",
+  value: any,
```
Comment (tvelu77) [inline src/front/src/pages/home/Home.tsx:20]: Why is it "any" ?
Shouldn't be "string" or JSON ?

File: src/front/src/sharedComponents/DisplayNotification.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                </Box>
+
+                <Box height={"3vh"} marginTop={1}>
+                    <Typography variant="body1">{"L'historique contient " + notificationList.length + " éléments."}</Typography>
+                </Box>
+            </Box>
+        </Drawer>
+        <SnackbarProvider maxSnack={3}>
+            <NotificationItems onNotificationReceived={message => {
+                saveNotification(message)
```
Comment (tvelu77) [inline src/front/src/sharedComponents/DisplayNotification.tsx:122]: Wouldn't that be heavy on the ressources ?
For example, if we've had 100 000 snackbars, would that be heavy for the front ?

Comment (Mamar-dev) [reply src/front/src/sharedComponents/DisplayNotification.tsx:122]: Except on a backend DDOS this should not be a problem. The queue is managed by the library

File: src/front/src/pages/home/Home.tsx (hunk 1 above)
Comment (Mamar-dev) [reply src/front/src/pages/home/Home.tsx:20]: Because

**Pull request merged**